### ISSUE TITLE
Improve Logging

### ISSUE THREAD

**Issue by christinebrass** [TrueCar/gluestick#461]
We should change the way we handle logging in GlueStick.  Right now, are [forcing use of Pino](https://github.com/TrueCar/gluestick/blob/ac23e6c762eb1497d91d2ea0a3e3beb6c6e9e97a/src/lib/server/logger.js#L6-L10) and that logger is unavailable to generated apps.

[Pino](https://github.com/pinojs/pino) was originally chosen as a simple solution for logging in JSON format.  We still have that requirement internally but perhaps GlueStick should actually take some sort of log provider and use that instead. By allowing the app to control the logger, we would be able to log statements directly from within the app and customize logged fields to the app's liking.

**Comment by michalchudziak** [comment]
I like the idea. Do you think that exposing `pino` to generated app would solve this issue, or do you want to let the user attach their own logger (maybe we could set `pino` as default option)?

**Comment by christinebrass** [comment]
I was originally thinking of solving it with the latter option (letting apps attach their own logger with `pino` as a default) but if exposing `pino` will allow generated apps more control, I'm good with trying that out.

**Comment by zamotany** [comment]
IMO we should postpone it until plugin system is established since it will be easier and we wouldn't have to rewrite it. I would also propose to replace Pino with [bunyan](https://github.com/trentm/node-bunyan), it's more flexible, has number of features that Pino doesn't have like multiple transports or option to specify transports depending on if it's info/warn/error etc, plus it's allow to use stream rotation which some users might be interested in.

**Comment by christinebrass** [comment]
I am OK with waiting until the plugin system is established 👍 . Would that automatically give us the option of specifying which logger to use within our apps?

I am open to replacing Pino with bunyan as the default GlueStick provider.  However, internally we are using some custom serializers with Pino, so if we were able to specify which logging provider to use that would make the transition easier. I do see that bunyan provides similar support for serializers so switching won't be tough if we had to do it.
